staging/src/k8s.io/apiserver/pkg/admission/interfaces.go

```diff
@@ -49,6 +49,23 @@ type Attributes interface {
 	GetKind() schema.GroupVersionKind
 	// GetUserInfo is information about the requesting user
 	GetUserInfo() user.Info
+
+	// AddAnnotation sets annotation according to key-value pair. The key should be qualified, e.g., podsecuritypolicy.admission.k8s.io/admit-policy, where
+	// "podsecuritypolicy" is the name of the plugin, "admission.k8s.io" is the name of the organization, "admit-policy" is the key name.
+	// An error is returned if the format of key is invalid. When trying to overwrite annotation with a new value, an error is returned.
+	// Both ValidationInterface and MutationInterface are allowed to add Annotations.
+	AddAnnotation(key, value string) error
+}
+
+// privateAnnotationsGetter is a private interface which allows users to get annotations from Attributes.
+type privateAnnotationsGetter interface {
+	getAnnotations() map[string]string
+}
+
+// AnnotationsGetter allows users to get annotations from Attributes. An alternate Attribute should implement
+// this interface.
+type AnnotationsGetter interface {
+	GetAnnotations() map[string]string
```

Comment:
why do we need this?

Reply:
I don't see this used. Premature abstraction? Would drop it.

Reply:
This interface is used according to @liggitt 's suggestion.

https://github.com/kubernetes/kubernetes/commit/d7474b875587c32b97f0cd93553ad1a9c9aa8441#r29220311

Reply:
It's used in WithAudit decorator to check the attribute interface.
Would be useful for alternate Attributes.

Reply:
ok, sounds reasonable.